src/shared/tests.c

```diff
@@ -301,3 +301,26 @@ int enter_cgroup_subroot(char **ret_cgroup) {
 int enter_cgroup_root(char **ret_cgroup) {
         return enter_cgroup(ret_cgroup, false);
 }
+
+const char *ci_environment(void) {
+        /* We return a string because we might want to provide multiple bits of information later on: not
+         * just the general CI environment type, but also whether we're sanitizing or not, etc. The caller is
+         * expected to use strstr on the returned value. */
+        static const char *ans = NULL;
+
+        if (ans)
+                return ans;
+
```

Comment:
shouldn't we check a generic env var first? i.e. CITYPE or so? and only fallback to the ugly checks if not set?

Reply:
$CI and $CONTINUOUS_INTEGRATION are set to "true", so they don't carry much useful information. Nothing sets $CITYPE. And I don't see much point for adding support for our own CITYPE variable that nothing else would use. For debugging, it's easy enough to set TRAVIS=true or some other variable as appropriate.

Reply:
well, but I think we should have the perspective of being able to drop this, i.e. move this out of the C code and into the CI build scripts, so that they just set the CITYPE env var if the CI systems don't set it themselves.

I mean, I sympathize that it's not trivial to update all the CI build scripts like this to set the CITYPE, but I still think it's the preferred way to do it like that, no? and thus we should put it first and most prominent.

I'd not just encode the status quo, but also where we want the journey to go